src/main/scala/org/virtuslab/internal/load/construct/YamlDecoder.scala

```diff
@@ -9,29 +8,110 @@ import scala.compiletime._
 import scala.compiletime.summonFrom
 import scala.deriving._
 import scala.util.Try
+import scala.language.implicitConversions
 
 sealed trait YamlDecoder[T]:
-  def construct(node: Node): Either[YamlError, T]
+  def construct(node: Node): Either[ConstructError, T]
 
 object YamlDecoder:
-  def apply[T](pf: PartialFunction[Node, Either[Throwable, T]]): YamlDecoder[T] =
+  def apply[T](pf: PartialFunction[Node, Either[ConstructError, T]]): YamlDecoder[T] =
     new YamlDecoder[T] {
-      override def construct(node: Node): Either[YamlError, T] =
+      override def construct(node: Node): Either[ConstructError, T] =
```

Comment:
Since `ConstructError` extends `YamlError`, it isn't necessary to change this type, is it?